nntrainer/tensor/tensor.cpp

```diff
@@ -336,9 +398,17 @@ Tensor Tensor::getSharedDataTensor(const TensorDim dim_,
       "Creating shared tensor of size bigger than tensor memory.");
 
   ret.dim = dim_;
-  ret.data = std::shared_ptr<float>(this->data, this->data.get() + offset);
   ret.strides = ret.dim.computeStrides();
 
+  /**
+   * In this case, its the caller's responsibility to ensure that allocate() is
+   * called for this function before operating on this tensor.
+   */
+  if (this->data)
+    ret.data = std::shared_ptr<float>(this->data, this->data.get() + offset);
+  else
```

Comment:
Is it possible to get the data from this->data.get()+offset1 to offset2?

Reply:
I don't understand the reference to offset1 and offset2 here. Can you please elaborate?

Reply:
What I means.. making shared tensor from in the middle of memory chunck.

Reply:
To remedy this, in the last commit, if the source tensor (say `t_s1`) provided depends on another source tensor (say `t_s2`), then the tensor to be made in this situation is made to directly depend on `t_s2`.